sdk/storage/inc/adls/protocol/adls_rest_client.hpp

```diff
@@ -0,0 +1,2220 @@
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/***************************************************************
+ * ALL THE CODE IN THIS FILE IS HANDWRITTEN, NOT AUTO-GENERATED.
+ **************************************************************/
+
+#pragma once
+
+#include <functional>
+#include <iostream>
+#include <map>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include "http/curl/curl.hpp"
+#include "http/http.hpp"
+#include "json.hpp"
+
+namespace Azure
+{
+namespace Storage
+{
+namespace ADLS
+{
+
+class ADLSRestClient
+{
+public:
+  constexpr static const char* k_DefaultServiceApiVersion = "2019-12-12";
```

Comment:
These common definitions and struct definitions and covert functions might be moved to `Model` namespace in the future, if we want to keep align with other language sdks. and it looks more concise to reference them as `Azure::Storage::ADLS::PathSetAccessControlRecursiveMode`.

So maybe they shouldn't be defined in ADLSRestClient?

Reply:
I think you are right. I realized that some data structures/enums in the protocol layer are going to be exposed to customers. I'll change this in the next iteration.